src/components/auth/EmailResetPassword.tsx

```diff
@@ -0,0 +1,130 @@
+"use client";
+
+import { useEffect, useInsertionEffect, useMemo, useState } from "react";
+import { useRouter, useSearchParams } from "next/navigation";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { useResetPasswordMutation } from "@/hooks/mutations/reset-password";
+import { useEmail } from "@/hooks/query/email";
+import { TextInput } from "../TextInput";
+import { Button } from "../ui/button";
+import { useToast } from "../ui/use-toast";
+import { Card } from "../Card";
+import { SkeletonEmailResetPassword } from "./SkeletonEmailResetPassword";
+
+const formSchema = z.object({
+  newPassword: z
+    .string()
+    .min(8, { message: "Password must be at least 8 characters" })
+    .regex(/\d/, { message: "Password must contain a number" })
+    .regex(/[a-z]/, { message: "Password must contain a lowercase letter" })
+    .regex(/[A-Z]/, { message: "Password must contain an uppercase letter" }),
+  confirmPassword: z
+    .string()
+    .min(8, { message: "Password must be at least 8 characters" })
+    .regex(/\d/, { message: "Password must contain a number" })
+    .regex(/[a-z]/, { message: "Password must contain a lowercase letter" })
+    .regex(/[A-Z]/, { message: "Password must contain an uppercase letter" })
+}).refine((data) => data.newPassword === data.confirmPassword, {
+  message: "Passwords do not match",
+  path: ["confirmPassword"],
+});
+
+export const EmailResetPassword = () => {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const { toast } = useToast();
+
+  const { data, isLoading } = useEmail(
+    searchParams.get("resetPasswordToken") ?? ""
+  );
+
+  const [formState, setFormState] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+  });
+
+  const { mutate, isPending } = useResetPasswordMutation(
+    searchParams.get("resetPasswordToken") ?? ""
+  );
+
+  const onSubmit = async (formData: z.infer<typeof formSchema>) => {
+    mutate(formData, {
+      onSuccess: () => {
+        setFormState(true)
+      },
+      onError: () => {
+        toast({
+          title: "Uh oh! Something went wrong",
+          description:
+            "An error occurred while creating your account. Please try again.",
+          variant: "destructive",
+        });
+      },
+    });
+  };
+  
+  let form
+  if (isLoading == true) {
```

Comment:
also try not to use `==` in javascript. can have some unintended consequences as it doesn't check type equality. always use `===`